### ISSUE TITLE
Miner panic in tests

### ISSUE THREAD

**Issue by quentinlesceller** [mimblewimble/grin#666]
Grin tests mining thread are panicking with:
```
thread 'miner' panicked at 'attempt to subtract with overflow', /Users/quentin/grin/core/src/global.rs:227:24
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:69
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:58
             at src/libstd/panicking.rs:381
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:397
   4: std::panicking::begin_panic
             at src/libstd/panicking.rs:577
   5: std::panicking::begin_panic
             at src/libstd/panicking.rs:538
   6: std::panicking::try::do_call
             at src/libstd/panicking.rs:522
   7: std::panicking::try::do_call
             at src/libstd/panicking.rs:498
   8: <core::ops::range::Range<Idx> as core::fmt::Debug>::fmt
             at src/libcore/panicking.rs:71
   9: <core::ops::range::Range<Idx> as core::fmt::Debug>::fmt
             at src/libcore/panicking.rs:51
  10: grin_core::global::difficulty_data_to_vector
             at /Users/quentin/grin/core/src/global.rs:227
  11: grin_core::consensus::next_difficulty
             at /Users/quentin/grin/core/src/consensus.rs:195
  12: grin_grin::miner::Miner::build_block
             at src/miner.rs:579
  13: grin_grin::miner::Miner::run_loop
             at src/miner.rs:475
  14: grin_grin::server::Server::start_miner::{{closure}}
             at src/server.rs:235
```

**Comment by quentinlesceller** [comment]
Looks like the issue appeared with https://github.com/mimblewimble/grin/pull/651.

**Comment by ignopeverell** [comment]
And the tests still succeed looks like...

**Comment by yeastplume** [comment]
Fixed via #669
